### ISSUE TITLE
New kinetic tuya device - Model ERC310

### ISSUE THREAD

**Issue by kvvoff** [arendst/Sonoff-Tasmota#6113]
Please add a new device to the wiki: AC dimmer ERC310. 

As well as [ERC309 ](https://github.com/arendst/Sonoff-Tasmota/issues/6056)supports kinetic switches without batteries. I just chose the configuration 54 Tuya Dimmer and can control the dimmer. 
The firmware process is described in [ERC309](https://github.com/arendst/Sonoff-Tasmota/issues/6056).

But I have a few problems, namely:
1. On the main page there are 6 switches. SetOption41 does not help. (SetOption41 0, SetOption41 1 .... SetOption41 8) nothing changes at all. They are always 6 switches

2. When I add this dimmer to the Home Assistant through mqtt autodiscovery (SetOption19 1) It also shows six different switches, among which only one turns this dimmer on and off. 
Ok, I make a command (SetOption30 1) [that is listed here](https://github.com/arendst/Sonoff-Tasmota/wiki/Home-Assistant#add-tasmota-devices-to-home-assistant) and in the assistant the light behaves generally strange: The first light just turns the dimmer on and off:
![1111](https://user-images.githubusercontent.com/38342171/61549032-b4943800-aa57-11e9-91cb-bc75248ff15e.png)

 And 6 light only changes the brightness of the slider, but DOES NOT allow the dimmer to be turned on or off:
![6](https://user-images.githubusercontent.com/38342171/61549049-c1189080-aa57-11e9-858f-d53769b4624f.png)

Very strange.

What am I doing wrong? :)

**Comment by meingraham** [comment]
@kvvoff,

It is the ERC309 that has the ESP chip which is flashed. The ERC310 is a package of the EC309 plus the kinetic switch; yes?

Is [this flashing guide](https://blakadder.github.io/templates/ERC309.html) accurate?

Mike

**Comment by kvvoff** [comment]
@meingraham It looks similar, but there is no relay
![image](https://user-images.githubusercontent.com/38342171/61553062-2ffae700-aa62-11e9-9ef8-5811ebb043b3.png)

**Comment by kvvoff** [comment]
And it also connects to kinetic switches and allows to adjust the brightness

**Comment by meingraham** [comment]
I may have the wrong impression of this device. I thought the ERC309 was just a controller and that it communicated with the switches which would have the relay circuitry.

**Comment by kvvoff** [comment]
309 is a relay, but it is not controlled by gpio, but through a separate MCU
310 is a dimmer, which is also (like all Tuya dimmers) controlled through a separate MCU

![444](https://user-images.githubusercontent.com/38342171/61609277-41a8de00-ac5e-11e9-9a27-238ffe9d7b7e.png)


The kinetic switch is a wireless switch (which looks like an Xiaomi Aqara) and it connects directly to the MCU (no matter what firmware is on esp8266)

kinetic switch (no battery need) with erc*** modules: 
![555](https://user-images.githubusercontent.com/38342171/61609285-466d9200-ac5e-11e9-89ce-9419b661e9ac.png)

Xiaomi Aqara(with battery):
![333](https://user-images.githubusercontent.com/38342171/61609396-aa905600-ac5e-11e9-8934-9927d016d5fe.jpg)

**Comment by ascillato2** [comment]
Hi,

Any news on this? Have you tried to make a template?

**Comment by ascillato** [comment]
Hi,

Any news on this? Have you tried to make a template?

**Comment by kvvoff** [comment]
@ascillato I have no changes, any help on this is welcome

**Comment by ascillato** [comment]
If the device has a separated mcu to control the dimmer, you need to scan the communication between the esp8266 (with original firmware) and that mcu in order to have all required commands.

Do you have a oscilloscope or digital analyzer?

**Comment by ascillato** [comment]
@kvvoff Have you tried latest Tasmota version from http://thehackbox.org/tasmota/020300/sonoff.bin

**Comment by kvvoff** [comment]
@ascillato I'll try.  I will write the result

**Comment by kvvoff** [comment]
@ascillato 
Thank you for help!
All wark with Tasmota **6.6.0.3** (http://thehackbox.org/tasmota/020300/sonoff.bin)

![image](https://user-images.githubusercontent.com/38342171/62348258-3333af80-b505-11e9-81c2-1030d8f0399c.png)

Please, add ERC310 Dimmer to blakadder.github.io

Thanks!

**Comment by kvvoff** [comment]
I rushed to conclusions.  In HA, the light behaves strangely - when I turn it on, it turns on, but the switch status goes off.  brightness below about 30 percent shines brighter than brightness 70, 80, 90 and 100%.

**Comment by ascillato** [comment]
That is not a Tasmota issue. That is the STATE topic you are looking. You have to put the same as you see in the console when you turn it on

**Comment by kvvoff** [comment]
Ok, I try

**Comment by ascillato2** [comment]
Hi, any news on this?

**Comment by kvvoff** [comment]
@ascillato2 
The slider on a small level shines brightly. The slider at the maximum level shines medium. And it's in Tasmota 6.6.0.3 . I recorded a video:
https://www.youtube.com/watch?v=U8se0a50XDA

**Comment by s-hadinger** [comment]
It looks like the Dimmer is inverted on this device. The fact that it's still medium when at maximum level is a side effect of Gamma correction. You can try temporarily setting `LedTable 0` and when at maximum level, it should be darker.

With PWM devices, you could choose PMWi to automatically invert the dimmer. On Tuya Serial device, I don't think we have this option. `pwm_inverted` is not available.

@arendst Theo any idea on how to do this properly?

**Comment by kvvoff** [comment]
@s-hadinger The fact is that **on firmware version 6.6.0.0 the slider works as it should**, but **6 switches appear** (I wrote about this above). 

On firmware version 6.6.0.3 there is one switch, but it works like a video.

**Comment by s-hadinger** [comment]
Ah ok. The thread is too long...

**Comment by kvvoff** [comment]
Any thoughts on how to combine the 6 switches into 1?

**Comment by ascillato2** [comment]
@kvvoff 

Hi, 

Have you tried latest version from development branch? (6.6.0.4)

**Comment by kvvoff** [comment]
@ascillato2 Hi, I will try the new version, I will write the result

**Comment by ascillato2** [comment]
Hi,

Any news on this?

**Comment by ascillato2** [comment]
Hi,

Any news on this?

**Comment by kvvoff** [comment]
Hi, I'm sorry I didn’t answer for a long time.  I don’t have the opportunity to check now, I will check on Monday

**Comment by ascillato2** [comment]
Ok, no problem. Let's wait until you have time. Thanks.

**Comment by kvvoff** [comment]
@ascillato 
Sonoff-Tasmota 6.6.0.6
Same. Slider to the maximum - the brightness is dim. Slider to the minimum - the brightness is even dimmer. A slider somewhere in the middle - the brightness is brighter. Is this exactly the case in the firmware? Maybe you need to make some settings?

**Comment by kvvoff** [comment]
@ascillato At the moment, I have found only one way out. I am using the release version (with 6 switches). Next I added to Home Assistant lights.yaml:

> `- platform: group
>   name: ERC310 GROUP TEST
>   entities:
>     - light.erc310_dimmer_6
>     - light.erc310_dimmer`

![image](https://user-images.githubusercontent.com/38342171/63321781-65794580-c32a-11e9-9f4b-d7e6be078d29.png)
![image](https://user-images.githubusercontent.com/38342171/63321784-69a56300-c32a-11e9-95b4-be1225c56ab9.png)
![image](https://user-images.githubusercontent.com/38342171/63321798-71650780-c32a-11e9-8c7c-07545314e597.png)


That is, I use the 1 and 6 switch together. Since only one slider works in one, only a button in the other. I merged them and now I can press the switch and adjust the brightness. But one moment, after turning on, the slider is in the very left position, although the light shines brightly. That is, the status of the slider is not updated with respect to the actual brightness
